src/spade/transformer/ContextualTemporalTraversal.java

```diff
@@ -0,0 +1,272 @@
+        private Double graphMinTime;
+        private Double graphMaxTime;
+        private static final String keyOutputFilename = "outputFilename",keyAnnotationName = "annotationName",keyOutputTime = "outputTime";
+
+        private Boolean outputTime;
+        private String outputFilename;
+        private String annotationName;
+
+        private BufferedWriter outputWriter;
+
+        @Override
+        public boolean initialize(String arguments) {
+                try {
+                        // Parse arguments with defaults from config file
+                        Map<String, String> argumentsMap = HelperFunctions.parseKeyValPairs(arguments);
+                        annotationName = ArgumentFunctions.mustParseNonEmptyString(keyAnnotationName, argumentsMap);
+                        outputFilename = ArgumentFunctions.mustParseNonEmptyString(keyOutputFilename, argumentsMap);
+                        outputTime = ArgumentFunctions.mustParseBoolean(keyOutputTime, argumentsMap);
+                        if (outputTime) {
+                                File file = new File(outputFilename);
+                                if (!file.exists()) {
+                                        file.createNewFile();
+                                }
+                                outputWriter = new BufferedWriter(new FileWriter(file, true));
+                                logger.log(Level.INFO, "Output file for traversal results will be written to: " + outputFilename);
+                        }
+                        // Initialize graphMinTime and graphMaxTime to null
+                        graphMinTime = null;
+                        graphMaxTime = null;
+                        
+                        return true;
+                } catch (Exception e) {
+                        logger.log(Level.SEVERE, "Failed to initialize ContextualTemporalTraversal", e);
+                        return false;
+                }
+        }
+
+        @Override
+        public LinkedHashSet<ArgumentName> getArgumentNames() {
+                return new LinkedHashSet<ArgumentName>(Arrays.asList(ArgumentName.SOURCE_GRAPH));
+        }
+
+        /*
+         * Check all edges that have child matching our target Vertex and see their timestamps
+         * Find edge with lowest timestamp that is our start time for our traversal
+         */
+        public Double getMintime(AbstractVertex vertex,HashMap<String, HashMap<String, HashMap<String, AbstractEdge>>> edgeMap) {
+                Double minTime = null;
+                HashMap<String, AbstractEdge> vertexParentEdges = edgeMap.get(vertex.bigHashCode()).get("parentEdges");
+
+                for (HashMap.Entry<String, AbstractEdge> entry : vertexParentEdges.entrySet()) {
+                    AbstractEdge edge = entry.getValue();
+                    AbstractEdge newEdge = createNewWithoutAnnotations(edge);
```

Comment:
Can you please share why `createNewWithoutAnnotations` is used here instead of getting the `time` from the `edge` itself? I don't have the context.

I am asking because the method `createNewWithoutAnnotations` creates a deep-copy of the `edge` here.

Reply:
Seems to be a mistake, I have fixed it.

Reply:
No worries :) If anything got left or got missed then we can take care of it in a future PR. Thank you.